core-api/src/main/java/com/optimizely/ab/internal/VoidCondition.java

```diff
@@ -14,13 +14,23 @@
  *    See the License for the specific language governing permissions and
  *    limitations under the License.
  */
+package com.optimizely.ab.internal;
 
-package com.optimizely.ab.config.audience.match;
+import com.optimizely.ab.config.ProjectConfig;
+import com.optimizely.ab.config.audience.Condition;
 
-public class UnknownMatchTypeException extends Exception {
-    private static String message = "uses an unknown match type. You may need to upgrade to a newer release of the Optimizely SDK";
+import javax.annotation.Nullable;
+import java.util.Map;
 
-    public UnknownMatchTypeException() {
-        super(message);
+/**
+ * Dummy condition that always evaluates to {@code null}
+ */
+public final class VoidCondition<T> implements Condition<T> {
+    static final VoidCondition INSTANCE = new VoidCondition();
```

Comment:
Maybe we can have different versions of VoidCondition that log [different warning messages](https://docs.google.com/document/d/1DfZzY-dReyu-hElaPngS849C6HC1wFXzuC_Ir0ShCdo/edit#heading=h.gc2x9d7t8f88) when they're evaluated?